challenges/2022-10-11-blast-from-the-past/setups/julia/solution.jl

```diff
@@ -0,0 +1,46 @@
+# # Find Three Largest Numbers
+# 
+# Write a function that takes in an array of at least three integers and, without 
+# sorting the input array, returns a sorted array of the three largest integers in
+# the input array. The function should return duplicate integers if necessary. 
+# 
+# ## Business Rules/Errata
+# 
+# - The input array should have at least three integers. If it does not, you should
+#   return a null value.
+# - You may not sort the input array
+# - The function should handle duplicate integers; for example `[10, 5, 9, 10, 12]`
+#   should return `[10, 10, 12]`
+# - Constant space -> you will return a new array of 3 integers, and this will be the
+#   only new data structure you create.
+# - Linear time -> you should solve this problem by only passing through the array a
+#   single time.
+
+function find_three_largest_numbers(arr::Vector{Int})::Vector{Int}
+    missing
+end
+
+# Tests ------------------------------------------------------------------------
+
+using Pkg; Pkg.add("Test"); using Test
+
+@testset "Test case one" begin
+    input    = [141, 2, 17, -7, -17, -27, 18, 541, 8, 7, 7]
+    result   = find_three_largest_numbers(input)
+    expected = [18, 141, 541]
+    @test all(result .== expected)
+end
+
+@testset "Test case two" begin
+    input    = [11, -7, 5, -7]
+    result   = find_three_largest_numbers(input)
+    expected = [-7, 5, 11]
+    @test all(result .== expected)
+end
+
+@testset "Test case three" begin
+    input    = [1]
+    result   = find_three_largest_numbers(input)
+    expected = [1]
```

Comment:
Should `expected` be `null` here? Or perhaps Typing is requiring `Vector{Int}`

From the instructions
```markdown
- The input array should have at least three integers. If it does not, you should
#   return a null value.
```

Reply:
I agree, can you not return an empty Vector{Int}?

Reply:
Well, I'll admit, I've missed this in _all_ the setups I've done. I just saw that the original challenge I copied these from returned a `[1]` for a `[1]` input and went with that. *But*, the challenge instructions definitely indicate a `null` (or equivalent) there. I'll update my setups, but we need to make sure that all the setups return `null` (or equivalent) for that test.

Reply:
Actually, I can't even blame the original version. The Java version checks for `null`, and I just misread it. D'oh!